src/com/google/enterprise/adaptor/DocumentHandler.java

```diff
@@ -1000,8 +1001,40 @@ private void transform() {
       Map<String, String> params = new HashMap<String, String>();
       params.put("DocId", docId.getUniqueId());
       params.put("Content-Type", contentType);
+      if (null != lastModified) {
+        params.put("Last-Modified-Millis-UTC", "" + lastModified.getTime());
+      }
+      String origDisplayUrl = null;
+      if (null != displayUrl) {
+        origDisplayUrl = "" + displayUrl;
+      }
+      params.put("Display-URL", origDisplayUrl);
+      params.put("Crawl-Once", "" + crawlOnce);
+      params.put("Lock", "" + lock);
       transform.transform(metadata, params);
       contentType = params.get("Content-Type");
+      try {
+        final String lmMillisUTC = params.get("Last-Modified-Millis-UTC");
+        if (!Strings.isNullOrEmpty(lmMillisUTC)) {
+          final long lm = Long.parseLong(lmMillisUTC);
+          if (lastModified.getTime() != lm) {
+            lastModified = new Date(lm);
+          }
+        }
+      } catch (NumberFormatException e) {
+        log.log(Level.FINER,
+            "Failed changing last-modified date {0}", params.get("Last-Modified-Millis-UTC"));
+      }
+      try {
+        final String du = params.get("Display-URL");
+        if (!Strings.isNullOrEmpty(du)) {
+          displayUrl = new URI(du);
```

Comment:
I added the equality check. (I did not remove the null, check because if the value will be null and you call .equal on null (you know what will happen) ;)
Also the JVM will optimize the routines regarding the null checks on multiple runs.